src/dyn/dyn_projectors.cpp

```diff
@@ -338,88 +338,87 @@ vector<int> get_stochastic_reordering2(CMATRIX& time_overlap, Random& rnd){
-    by some other index, j = perm[i]. 
+    chosen_permutation - list of integers that describe the permutation. That is:
+    chosen_permutation[i] - is the index identifying the "older" state "i". Now, it may be labeled
+    by some other index, j = chosen_permutation[i].
 
     |phi_j(t+dt)> = a_0j |phi_0(t)> + a_1j |phi_1(t)> + ... a_Nj |phi_N(t)>
 
     So:  <phi_i(t)|phi_j(t+dt)> = a_ij and then
- 
+
     |a_ij|^2 is the probability that the state i at time t becomes state j at time t+dt
 
     i - state index at t
-    perm[i] - state index at t+dt
+    chosen_permutation[i] - state index at t+dt
 
     """
     */
 
-    // extract the indices where <phi_i(t)|phi_i(t+dt)> is not close to 1. 
+    // extract the indices where <phi_i(t)|phi_i(t+dt)> is not close to 1.
     CMATRIX S(time_overlap);  // just a temporary working object
-    int sz = time_overlap.n_rows;
-    int i,j, target;
-
-    // Original permutation
-    vector<int> perm(sz, 0);  
-    for(i=0;i<sz;i++){ perm[i] = i; } 
-
-
-    vector<int> used;  // already utilized options
-    MATRIX g(1, sz);   // switching probabilities
 
-    // was sz - 1 - not sure why
-    for(i=0;i<sz;i++){ // all starting states (at time t)
-
-        // Make a list of the states that can still participate
-        // in the reordering
-        vector<int> active_states;
-        for(j=0;j<sz;j++){ 
-            if(is_in_vector(j,used)){  ;;  }
-            else{ active_states.push_back(j); }
-        }
-
-
-        int act_st_sz = active_states.size(); // should be equal to sz - i
-
-        if(act_st_sz > 1){
- 
-            MATRIX g(1, act_st_sz);
-            for(j=0; j<act_st_sz; j++){ // all target states
-
-                int indx = active_states[j];
-                g.set(0,j, (S.get(i,indx)*std::conj(S.get(i,indx))).real() );
-            }
- 
-            // Now stochastically determine the states re-assignment
-            double ksi = rnd.uniform(0.0, 1.0);
-
-            target = hop(0, g, ksi);    // internal (active states') state index
-            if(target > act_st_sz-1){
-                cout<<"ERROR: target = "<<target<<" is larger than max possible value = "<<act_st_sz-1<<endl;
-                exit(0);
-            }
-            target = active_states[target]; // actual state index
-        }
-        else if(act_st_sz == 1){
-            target = active_states[0];
-        }
-
-        // Update the permutaiton and add the found target state to the exclude list
-        perm[i] = target;
-        used.push_back(target);
-
-    }// for i
+    int initial_state = 0;
+    double ksi = rnd.uniform(0.0, 1.0);
+    int number_states = time_overlap.n_rows;
+    vector<int> list_states;
+    vector<int> permutation;
+    vector<double> all_probablities;
+    double permutation_probability;
+    double sum_overlap;
+    double state_probability;
+
+// Construct perumtations and calculate probability of each permutation
+
+    for(int i = 0; i<number_states; i++)

```

Comment:
style comment: if you use the whitespaces around the operator "=", you may also add them around "<" operator too, to have the same style in both places.  "i++" stays the same though